### PR TITLE
`bslib::sidebar()` tests

### PR DESCRIPTION
**Pull request by cpsievert**
Covers the various ways a `sidebar()` can be rendered with `layout_sidebar()`, `page_navbar()`, and `navs_tab_card()`/`navs_pill_card()`

### CODE REVIEW

File: inst/apps/308-sidebar-kitchen-sink/app.R [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  ),
+  nav(
+    "Fill",
+    plot_price,
+    br(),
+    layout_sidebar(side_left, plot_price, fillable = TRUE),
+    br(),
+    card(
+      card_header("Depth"),
+      card_sidebar(side_left, plot_depth, fillable = TRUE)
```
Comment (cpsievert) [inline inst/apps/308-sidebar-kitchen-sink/app.R:52]:
```suggestion
      layout_sidebar(side_left, plot_depth, fillable = TRUE)
```

File: inst/apps/308-sidebar-kitchen-sink/app.R [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+  ),
+  nav(
+    "Fill+Scroll",
+    plot_price,
+    br(),
+    layout_sidebar(side_left, plot_price),
+    br(),
+    card(
+      card_header("Depth"),
+      card_sidebar(side_left, plot_depth)
```
Comment (cpsievert) [inline inst/apps/308-sidebar-kitchen-sink/app.R:63]:
```suggestion
      layout_sidebar(side_left, plot_depth)
```

File: inst/apps/308-sidebar-kitchen-sink/app.R [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+  ),
+  nav(
+    "Scroll",
+    plot_price,
+    br(),
+    layout_sidebar(side_left, plot_price),
+    br(),
+    card(
+      card_header("Depth"),
+      card_sidebar(side_left, plot_depth)
```
Comment (cpsievert) [inline inst/apps/308-sidebar-kitchen-sink/app.R:74]:
```suggestion
      layout_sidebar(side_left, plot_depth)
```

File: inst/apps/308-sidebar-kitchen-sink/tests/testthat/test-shinytest2.R [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  # TODO: why isn't this working?
+  app$set_inputs(inner_sidebar = TRUE)
+  app$expect_screenshot()
+
+  app$click("toggle_sidebar")
+  app$set_inputs(inner_sidebar = FALSE)
+  app$set_inputs(outer_sidebar = FALSE)
+  app$expect_screenshot()
+
+  # TODO: now test that fill doesn't apply on mobile
```
Comment (cpsievert) [inline inst/apps/308-sidebar-kitchen-sink/tests/testthat/test-shinytest2.R:39]: Still need to add tests for mobile

**Comment by schloerke** [comment]
LGTM; Waiting mobile tests from https://github.com/rstudio/shinycoreci/pull/153#discussion_r1132597042

**Comment by cpsievert** [comment]
@gadenbuie these are the only other sidebar tests we have at the moment (besides #164)...lmk if you think there's important scenarios that we need coverage on...

**Review by gadenbuie — APPROVED** [review]
I think these are good smoke tests and a good compliment to our other sidebar tests, each of which are more fine-grained. I don't see anything that we need to add to this test at this time

**Pull request merged**